ubccr/xdmod
# Updated enum_user_types_and_roles to utilize acls

## Description
Update Internal Dashboard -> Existing Users -> Role Filter ( having a role of <role> ) to utilize Acls. This also entailed updating the backend for the Internal Dashboard -> User Management -> Existing Users -> Prepare E-Mail button. Specifically, the Role Filter used to determine the list of users to send the email to. Also cleaned up this function `deriveUserEnumerationQuery` to remove unused parameters ( `context_filter` ) and change it from Dynamic SQL to a parameterized query. 

## Motivation and Context
The Role Filtering was not operating as intended due to roles having difference id's than acls ( perhaps obviously )

## Tests performed
**Manually Tested:**
- Log in to the Internal Dashboard
- Click the User Management -> Existing Users tab
- Select the 'having a role of: <Role>' drop down
- Select a number of different roles
- Ensure that the users displayed do indeed have the selected role
- Click the 'Prepare E-Mail' toolbar button
- Select the 'having a role of: <Role>' drop down
- Select a role
- Click the 'Send' button
- Ensure that the users identified by the email addresses displayed have the role previously selected.

**Integration Tests:**
`ControllerTest::testEnumUserTypesAndRoles`



## Types of changes


## Checklist:
